### PR TITLE
feat: add on change handlers and renamed others for standardisation

### LINKED ISSUE #2119: Components: add onChange-like callback
## Bug Report
Some components do not have the proper `onChange` type callbacks. For example `Menu` does not have `onActiveIndexChange`. Also Tree etc.

### Expected Result
Components should have these callbacks for the controlled props.

### PR DESCRIPTION
**Pull request by silviuaavram**
# BREAKING CHANGES

Some onChange handlers have been renamed to become standardised with HTML elements and with the name of the prop they are representing. The list of changes:

| Component | Old Handler Name | New Handler Name |
| ------------ | ------------------ | -------------------- |
| Alert  | onDismiss  | onVisibleChange |
| Dropdown  | onSelectedChange  | onChange |
| Embed  | onActiveChanged  | onActiveChange |
| RadioGroup  | checkedValueChanged  | onCheckedValueChange |
| RadioGroupItem  | checkedChanged  | onChange |

Some additional handlers added:

| Component | Handler Name |
| ------------ |--------------- |
| Accordion | onActiveIndexChange |
| Dropdown  | onActiveSelectedIndexChange |
| Dropdown  | onHighlightedIndexChange |
| Menu  | onActiveIndexChange |
| ToolbarMenuItem  | onMenuOpenChange |
| Tree  | onActiveItemIdsChange |

isConformat tests have been improved. If a component has a state value as controlled prop, then it should also have a default prop and a onChange handler as handled props.

Fixes https://github.com/microsoft/fluent-ui-react/issues/2119

Some components could not be covered by the test as they are not conformant (Dialog, Tooltip, Popup etc.)

### CODE REVIEW

File: packages/react/test/specs/components/Tree/Tree-test.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ const checkOpenTitles = (wrapper: ReactWrapper, expected: string[]): void => {
 }
 
 describe('Tree', () => {
-  isConformant(Tree)
+  isConformant(Tree, { autocontrolledPropMappings: { activeItemIds: 'onActiveItemIdsChange' } })
```
Comment (mnajdova) [inline packages/react/test/specs/components/Tree/Tree-test.tsx]: Why not just adding a list of auto controlled props?

File: packages/react/test/specs/commonTests/isConformant.tsx [hunk 2]
```diff
@@ -34,6 +34,8 @@ export interface Conformant {
   /** This component uses wrapper slot to wrap the 'meaningful' element. */
   wrapperComponent?: React.ReactType
   handlesAsProp?: boolean
+  /** In the case when a onChange prop name is custom in relation to the controlled prop. */
```
Comment (jdhuntington) [inline packages/react/test/specs/commonTests/isConformant.tsx]: Comment is confusing; the way I read the implementation it seems like this key is always required, not just when the prop name is customized.

Comment (silviuaavram) [reply packages/react/test/specs/commonTests/isConformant.tsx]: oups, leftover from previous implementation, will correct that.

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -432,7 +454,7 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
     const { highlightedIndex, open, searchQuery, value } = this.state
 
     return (
-      <ElementType className={classes.root} {...unhandledProps}>
+      <ElementType className={classes.root} onChange={this.handleChange} {...unhandledProps}>
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/Dropdown.tsx:458]: Aren't we going to end up here with `onChange` handler on a `div` element?

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -1272,7 +1311,7 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
    * because we want to keep the event handling interface
    */
   setStateAndInvokeHandler = (
-    handlerName: keyof DropdownProps,
+    handlerNames: string[],
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/Dropdown.tsx]: Let's make this array of `keyof DropdownProps`.

File: packages/react/src/components/Menu/Menu.tsx [hunk 5]
```diff
@@ -144,21 +152,26 @@ class Menu extends AutoControlledComponent<WithAsProp<MenuProps>, MenuState> {
   static Item = MenuItem
   static Divider = MenuDivider
 
+  setActiveIndex = (e: React.SyntheticEvent, activeIndex: number) => {
+    _.invoke(this.props, 'onActiveIndexChange', e, this.props)
```
Comment (mnajdova) [inline packages/react/src/components/Menu/Menu.tsx]: Let's send here the new `activeIndex` together with the props?

File: packages/react/src/components/RadioGroup/RadioGroup.tsx [hunk 6]
```diff
@@ -200,7 +200,7 @@ class RadioGroup extends AutoControlledComponent<WithAsProp<RadioGroupProps>, an
     props: RadioGroupItemProps
   }) {
     this.setState({ checkedValue, shouldFocus })
-    _.invoke(this.props, 'checkedValueChanged', event, props)
+    _.invoke(this.props, 'onCheckedValueChange', event, props)
```
Comment (mnajdova) [inline packages/react/src/components/RadioGroup/RadioGroup.tsx:203]: Shouldn't this be `onChange`?

File: packages/react/src/components/Toolbar/ToolbarMenuItem.tsx [hunk 7]
```diff
@@ -90,6 +90,13 @@ export interface ToolbarMenuItemProps
    */
   onClick?: ComponentEventHandler<ToolbarMenuItemProps>
 
+  /**
+   * Called when the menu inside the item opens or closes.
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props, with `menuOpen` reflecting the new state.
+   */
+  onMenuOpenChange?: ComponentEventHandler<ToolbarMenuItemProps>
```
Comment (mnajdova) [inline packages/react/src/components/Toolbar/ToolbarMenuItem.tsx:98]: Shouldn't we invoke this somewhere?

File: packages/react/src/components/Tree/Tree.tsx [hunk 8]
```diff
@@ -281,14 +288,20 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
...
-        activeItemIds,
-      })
+      this.setActiveItemIds(e, activeItemIds)
 
       _.invoke(predefinedProps, 'onSiblingsExpand', e, treeItemProps)
     },
   })
 
+  setActiveItemIds = (e: React.SyntheticEvent, activeItemIds: string[]) => {
+    _.invoke(this.props, 'onActiveItemIds', e, this.props)
```
Comment (mnajdova) [inline packages/react/src/components/Tree/Tree.tsx]: Send the updated `activeItemIds` please together with the props.

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 9]
```diff
@@ -154,6 +154,20 @@ export interface DropdownProps
   /** A message to be displayed in the list when the dropdown has no items. */
   noResultsMessage?: ShorthandValue<ListItemProps>
 
+  /**
+   * Called when the dropdown's selected items index change.
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props and the new selected value(s).
+   */
+  onActiveSelectedIndexChange?: ComponentEventHandler<DropdownProps>
```
Comment (layershifter) [inline packages/react/src/components/Dropdown/Dropdown.tsx:163]:
```suggestion
  onActiveSelectedIndexChange?: ComponentEventHandler<DropdownProps>
```

It's aligned, but name looks really strange 💣

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (layershifter) [reply packages/react/src/components/Dropdown/Dropdown.tsx]: Actually `(keyof DropdownProps)[]`

File: packages/react/src/components/Embed/Embed.tsx [hunk 10]
```diff
@@ -44,7 +44,7 @@ export interface EmbedProps extends UIComponentProps {
    * @param event - React's original SyntheticEvent.
    * @param data - All props and proposed value.
    */
-  onActiveChanged?: ComponentEventHandler<EmbedProps>
+  onActiveChange?: ComponentEventHandler<EmbedProps>
```
Comment (layershifter) [inline packages/react/src/components/Embed/Embed.tsx:47]: 👍

File: packages/react/src/components/Toolbar/ToolbarMenuItem.tsx (hunk 7 above)
Comment (layershifter) [reply packages/react/src/components/Toolbar/ToolbarMenuItem.tsx:98]: `onMenuOpenChange` vs `onMenuOpen` 😭

**Review by layershifter — COMMENTED** [review]
I wonder, to we want to include `Dialog`, `Popup` & `Tooltip` there?

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (silviuaavram) [reply packages/react/src/components/Dropdown/Dropdown.tsx:458]: I did the same thing as in RadioGroupItem. Just to pass the isConformant. I don't know any better way.

```
    // RadioGroupItem component doesn't present any `input` component in markup, however all of our
    // components should handle events transparently.
```

File: packages/react/src/components/RadioGroup/RadioGroup.tsx (hunk 6 above)
Comment (silviuaavram) [reply packages/react/src/components/RadioGroup/RadioGroup.tsx:203]: If the prop was `value`, then yes. Are we changing everything here?

File: packages/react/src/components/Toolbar/ToolbarMenuItem.tsx (hunk 7 above)
Comment (silviuaavram) [reply packages/react/src/components/Toolbar/ToolbarMenuItem.tsx:98]: What do you mean? Isn't it invoked on line 244?

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 9 above)
Comment (silviuaavram) [reply packages/react/src/components/Dropdown/Dropdown.tsx:163]: :(

File: packages/react/src/components/Toolbar/ToolbarMenuItem.tsx (hunk 7 above)
Comment (mnajdova) [reply packages/react/src/components/Toolbar/ToolbarMenuItem.tsx:98]: Ah got it, we are fixing typings, not adding new handler. Sorry for the coonfusion

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 11]
```diff
@@ -155,6 +155,20 @@ export interface DropdownProps
...
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/Dropdown.tsx:163]: Can we suggest in the comment if the event is null for some handler?

Comment (silviuaavram) [reply packages/react/src/components/Dropdown/Dropdown.tsx:163]: As far as I see, the event `might` be `null`, depending from which place the handler was called. And not only for this one, also for `onChange` as well. What do you suggest?

Comment (mnajdova) [reply packages/react/src/components/Dropdown/Dropdown.tsx:163]: But we already have an issue regarding this: https://github.com/microsoft/fluent-ui-react/issues/2318 We should distinguish between the component event handlers that have event and the ones which do not. At least this is my opinion, we can ask other people for input

File: packages/react/src/components/RadioGroup/RadioGroup.tsx (hunk 6 above)
Comment (mnajdova) [reply packages/react/src/components/RadioGroup/RadioGroup.tsx:203]: I see, I thought it is the native value prop, but I see now that this is the `RadioGroup` component.

File: packages/react/src/components/Embed/Embed.tsx [hunk 12]
```diff
@@ -124,7 +124,7 @@ class Embed extends AutoControlledComponent<WithAsProp<EmbedProps>, EmbedState>
 
     if (iframeNil || (!iframeNil && !this.state.active)) {
       this.setState({ active: !this.state.active })
-      _.invoke(this.props, 'onActiveChanged', e, { ...this.props, active: !this.state.active })
+      _.invoke(this.props, 'onActiveChange', e, { ...this.props, active: !this.state.active })
```
Comment (mnajdova) [inline packages/react/src/components/Embed/Embed.tsx]: Can you save the value that we want to add in the state in a variable and reuse it in both setState and here? This is not save, as setState is async, so we should either move this into the callback in setState, or add here a variable's value, not the state's value

**Review by mnajdova — APPROVED** [review]

**Pull request merged**